cpp/src/quantiles/quantiles.cu

```diff
@@ -16,196 +16,253 @@
 
 //Quantile (percentile) functionality
 
-#include "quantiles.h"
+#include "quantiles.hpp"
 #include "utilities/cudf_utils.h"
 #include "utilities/error_utils.hpp"
 #include "utilities/type_dispatcher.hpp"
+#include "utilities/wrapper_types.hpp"
 #include "rmm/thrust_rmm_allocator.h"
 #include "cudf.h"
 
 #include <thrust/device_vector.h>
 #include <thrust/copy.h>
 
-namespace{ //unknown
-  template<typename VType,
-           typename RetT = double>
-    void f_quantile_tester(rmm::device_vector<VType>& d_in)
+
+namespace{ // anonymouys namespace
+
+  struct QuantiledIndex {

```

Comment:
I think this should be called `QuantileIndex` (note no 'd'). 

Also, what is our style rule for capitalizing struct names? Is it `QuantileIndex` or `quantile_index`?